hooks/queries/tokenMetadata.ts

```diff
@@ -0,0 +1,40 @@
+import { findMetadataPda } from '@metaplex-foundation/js'
+import { Metadata } from '@metaplex-foundation/mpl-token-metadata'
+import { PublicKey } from '@solana/web3.js'
+import { useQuery } from '@tanstack/react-query'
+import useWalletStore from 'stores/useWalletStore'
+
+export const useTokenMetadata = (
+  mint: PublicKey | undefined,
+  otherConditions?: boolean
+) => {
+  const connection = useWalletStore((s) => s.connection)
+
+  const enabled = !!mint && !!otherConditions
+
+  const query = useQuery({
```

Comment:
is token metadata even mutable generally? 

we can do `useQuery({cacheTime: (10 days in miliseconds), staleTime: (1hr in miliseconds)` for example, and then it will just get cached a long time and not get refreshed more than each hour